app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -142,12 +142,24 @@ def create
       unless repo_params[:gpg_key_id].blank?
         gpg_key = @gpg_key
       end
+      repo_params[:arch] = repo_params[:arch] || 'noarch'
       repo_params[:label] = labelize_params(repo_params)
       repo_params[:url] = nil if repo_params[:url].blank?
-      unprotected =  repo_params.key?(:unprotected) ? repo_params[:unprotected] : true
-      repository = @product.add_repo(repo_params[:label], repo_params[:name], repo_params[:url],
-                                     repo_params[:content_type], unprotected,
-                                     gpg_key, repository_params[:checksum_type], repo_params[:download_policy])
+      unprotected = repo_params.key?(:unprotected) ? repo_params[:unprotected] : true
+      repo_param = {
+        :label => repo_params[:label],
+        :name =>  repo_params[:name],
+        :url => repo_params[:url],
+        :content_type => repo_params[:content_type],
+        :arch => repo_params[:arch],
+        :unprotected => unprotected,
+        :gpg_key => gpg_key,
+        :checksum_type => repository_params[:checksum_type],
+        :download_policy => repo_params[:download_policy]
+
+      }
+
+      repository = @product.add_repo(repo_param)
```

Comment:
since most of the params are coming from the request, I'd probably prefer:
```
repository = @product.add_repo(params.slice(:name, :label, :url, :content_type, :arch, :download_policy, :checksum_type, :gpg_key, :unprotected ))
```

you might have to set :gpg_key and :unprotected on the repo_params hash before this, but no need for an additional hash